flow/connectors/postgres/postgres.go

```diff
@@ -1011,13 +1011,19 @@ func (c *PostgresConnector) AddTablesToPublication(req *protos.AddTablesToPublic
 			return fmt.Errorf("some additional tables not present in custom publication: %s",
 				strings.Join(notPresentTables, ", "))
 		}
+	} else {
+		for _, additionalSrcTable := range additionalSrcTables {
+			_, err := c.conn.Exec(c.ctx, fmt.Sprintf("ALTER PUBLICATION %s ADD TABLE %s",
+				c.getDefaultPublicationName(req.FlowJobName), additionalSrcTable))
+			// don't error out if table is already added to our publication
+			if err != nil && !strings.Contains(err.Error(), "SQLSTATE 42710") {
```

Comment:
Should be able to avoid string comparisons using https://pkg.go.dev/github.com/jackc/pgerrcode

Reply:
isn't granular enough, both invalid publication name and invalid table name come under class 42 as well